field/eisenstein/eisenstein_test.go

```diff
@@ -0,0 +1,278 @@
+		func(a *ComplexNumber) bool {
+			var b ComplexNumber
+			b.Neg(a)
+			a.Neg(a)
+			return a.Equal(&b)
+		},
+		genE,
+	))
+
+	properties.Property("Having the receiver as operand (conjugate) should output the same result", prop.ForAll(
+		func(a *ComplexNumber) bool {
+			var b ComplexNumber
+			b.Conjugate(a)
+			a.Conjugate(a)
+			return a.Equal(&b)
+		},
+		genE,
+	))
+
+	properties.TestingRun(t, gopter.ConsoleReporter(false))
+}
+
+func TestEisensteinArithmetic(t *testing.T) {
+
+	t.Parallel()
+	parameters := gopter.DefaultTestParameters()
+	if testing.Short() {
+		parameters.MinSuccessfulTests = nbFuzzShort
+	} else {
+		parameters.MinSuccessfulTests = nbFuzz
+	}
+
+	properties := gopter.NewProperties(parameters)
+
+	genE := GenComplexNumber()
+
+	properties.Property("sub & add should leave an element invariant", prop.ForAll(
+		func(a, b *ComplexNumber) bool {
+			var c ComplexNumber
+			c.Set(a)
+			c.Add(&c, b).Sub(&c, b)
+			return c.Equal(a)
+		},
+		genE,
+		genE,
+	))
+
+	properties.Property("neg twice should leave an element invariant", prop.ForAll(
+		func(a *ComplexNumber) bool {
+			var b ComplexNumber
+			b.Neg(a).Neg(&b)
+			return a.Equal(&b)
+		},
+		genE,
+	))
+
+	properties.Property("conj twice should leave an element invariant", prop.ForAll(
+		func(a *ComplexNumber) bool {
+			var b ComplexNumber
+			b.Conjugate(a).Conjugate(&b)
+			return a.Equal(&b)
+		},
+		genE,
+	))
+
+	properties.Property("add zero should leave element invariant", prop.ForAll(
+		func(a *ComplexNumber) bool {
+			var b ComplexNumber
+			zero := new(ComplexNumber)
+			b.Add(a, zero)
+			return a.Equal(&b)
+		},
+		genE,
+	))
+
+	properties.Property("mul by one should leave element invariant", prop.ForAll(
+		func(a *ComplexNumber) bool {
+			var b ComplexNumber
+			one := &ComplexNumber{
+				*big.NewInt(1),
+				*big.NewInt(0),
+			}
+			b.Mul(a, one)
+			return a.Equal(&b)
+		},
+		genE,
+	))
+
+	properties.Property("add should be commutative", prop.ForAll(
+		func(a, b *ComplexNumber) bool {
+			var c, d ComplexNumber
+			c.Add(a, b)
+			d.Add(b, a)
+			return c.Equal(&d)
+		},
+		genE,
+		genE,
+	))
+
+	properties.Property("add should be assiocative", prop.ForAll(
+		func(a, b, c *ComplexNumber) bool {
+			var d, e ComplexNumber
+			d.Add(a, b).Add(&d, c)
+			e.Add(c, b).Add(&e, a)
+			return e.Equal(&d)
+		},
+		genE,
+		genE,
+		genE,
+	))
+
+	properties.Property("mul should be commutative", prop.ForAll(
+		func(a, b *ComplexNumber) bool {
+			var c, d ComplexNumber
+			c.Mul(a, b)
+			d.Mul(b, a)
+			return c.Equal(&d)
+		},
+		genE,
+		genE,
+	))
+
+	properties.Property("mul should be assiocative", prop.ForAll(
+		func(a, b, c *ComplexNumber) bool {
+			var d, e ComplexNumber
+			d.Mul(a, b).Mul(&d, c)
+			e.Mul(c, b).Mul(&e, a)
+			return e.Equal(&d)
+		},
+		genE,
+		genE,
+		genE,
+	))
+
+	properties.Property("norm should always be positive", prop.ForAll(
```

Comment:
Norm can be zero for `0 + 0*\omega`? In that case imo the comparison below should be `a.Norm().Sign() >= 0`